src/main/webapp/app/orion/management/orion-programming-exercise.component.ts

```diff
@@ -0,0 +1,40 @@
+import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';
+import { ProgrammingExercise } from 'app/entities/programming-exercise.model';
+import { OrionConnectorService } from 'app/shared/orion/orion-connector.service';
+import { ExerciseView, OrionState } from 'app/shared/orion/orion';
+import { Router } from '@angular/router';
+import { Course } from 'app/entities/course.model';
+
+@Component({
+    selector: 'jhi-orion-programming-exercise',
+    templateUrl: './orion-programming-exercise.component.html',
+})
+export class OrionProgrammingExerciseComponent implements OnInit {
+    @Input() embedded = false;
+    @Input() course: Course;
+    @Output() exerciseCount = new EventEmitter<number>();
+
+    readonly ExerciseView = ExerciseView;
+
+    orionState: OrionState;
+
+    constructor(private orionConnectorService: OrionConnectorService, private router: Router) {}
+
+    ngOnInit() {
+        this.orionConnectorService.state().subscribe((state) => {
+            this.orionState = state;
+        });
+    }
+
+    editInIDE(programmingExercise: ProgrammingExercise) {
+        this.orionConnectorService.editExercise(programmingExercise);
+    }
+
+    openOrionEditor(exercise: ProgrammingExercise) {
+        try {
+            this.router.navigate(['code-editor', 'ide', exercise.id, 'admin', exercise.templateParticipation?.id]);
+        } catch (error) {
+            this.orionConnectorService.log(error);
+        }
```

Comment:
What is the reason this navigation can fail? Could you maybe add a comment for it?

Reply:
I have to admit I have no idea an have never seen any failures. I don't really feel comfortable with deleting it either though, assuming whoever put it there probably had their reasons.

Reply:
I would rather haven an additional check than an error which does not get caught :D